### PR TITLE
LPS-83857 If the default localization doesn't have a reminder query, …

### PR DESCRIPTION
**Pull request by ChrisKian**
https://issues.liferay.com/browse/LPS-83857

Looking at [this article, specifically item #2 here](https://grow.liferay.com/share/DXP-+How+to+code+upgrade+process+and+backport+rules#section-DXP-+How+to+code+upgrade+process+and+backport+rules-Master), I believe my upgrade process is in the correct spot.  I've ran it to verify it works properly from where it is.  It's also safe to be ran on any data; running it multiple times or on data which doesn't need to be fixed won't hurt anything.

As far as the fix goes, it's pretty straight-forward.  We get the preferences from PortalPreferences if it's an org's preferences and it contains "reminderQueries", which is the we store the reminder query values under.  Assuming those criteria are met, we determine the org's default locale using it's companyId, append it to "reminderQueries_", and make sure it doesn't exist in the preferences (because then there is no issue to resolve).  If it doesn't exist, we enter the convertDefaultReminderQueries() method, which finds the default property "reminderQueries", copies it, changes the name to the localized value, and inserts it into the preferences.  The preferences are then updated, and the issue is resolved.

Let me know if you have any questions.  Thanks!

### CODE REVIEW

**Comment by ChrisKian** [comment]
ci:test:sf

**Comment by ChrisKian** [comment]
ci:test:relevant

**Comment by ChrisKian** [comment]
ci:test:upgrade

**Comment by ChrisKian** [comment]
Hey @drewbrokke,
I sent you two PRs yesterday, both of which were upgrade processes and ci:test:upgrade was ran on both.  They both have failures, but I don't think they are related to my PRs since they are duplicate and both PRs touch different areas (one is a fix for the org logo and the other modified portalPreferences).  Whenever you get to reviewing them, let me know if you want me to look at the failures with you.

TL;DR, I believe the upgrade failures aren't related.

**Comment by drewbrokke** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by drewbrokke** [comment]
Hey @ChrisKian, thanks for waiting while I got to this pull.  Overall this solution looks good to me, but we've also been trying to include regression testing with issues like this one. Since this one doesn't deal with a change in DB structure, it should be much easier to write a test case.  Would you be willing to re-send with a test case?  Thanks!
